### DISCUSSION TITLE
Concerned of Serverless Costs (AWS Lambda)

### DISCUSSION THREAD

**Discussion by tembra** [filamentphp/filament#10406] [General]
## Hello everyone!

I'm new here and I'm about to start a new SaaS project and deciding between Filament and Nova.

## Introduction

While Filament uses Livewire to render pages, Nova uses Vue. This make a huge difference in the data transmitted between server and client.

Javascript SPA is known to have a huge initial workload by downloading a bundle of javascript. But it's generally a single request from a **STATIC** server with an acceptable size (or at least should be 😛). Subsequent requests are almost only to an API.

> The introduction of modules seems that will solve the initial workload problem soon.

Livewire in other hand needs to handle the data (it seems to be a lot) from a **DYNAMIC** server (PHP) just to update the UI.

We all know that static resources are cheaper than dynamic ones.

Said that, I'm really concerned about the costs of a Filament app running on AWS Lambda where we pay for seconds while a lambda is up and running. Mainly after analyzing the demo apps and getting the results below.

## Question

**So anyone have any experience about this subject that can share?**

> I'm so in love with Filament that I want to use it 🥰 but I also need to take care of costs involving the lifecycle of the project.

## Comparison between Filament and Nova

> [!NOTE]
> The requests are filtered by URL

### Filament

Data collected from a full refresh on a simple form from demo URL: https://demo.filamentphp.com/shop/customers/create

![image](https://github.com/filamentphp/filament/assets/6340360/9087d9f9-05b1-4d0b-9e6a-33db3e6a934d)

### Nova

Data collected from a full refresh on a simple form from demo URL: https://nova-demo.laravel.com/nova/resources/events/new

![image](https://github.com/filamentphp/filament/assets/6340360/7a41f3df-00a6-415d-978b-b71f01c7cfb3)

### Comparison

Despite that Nova had a 1.09 MB of data transferred including assets (the why already stated above) it only took 876 ms to finish. And more important is that `creation-fields` request transferred only 2.38 kB in 233 ms.

In turn, Filament transferred only 205.13 kB but took 1.67 s to finish. And `update` request from Livewire transferred 5.34 kB in 896 ms.

Analyzing only the mentioned requests we can stated that Filament transferred 2,24x more data and uses 3,84x more time than Nova.

I know that I must take in consideration that the apps are not in the same server with same structure. But I doubt that this huge difference is only by that and risk to say that server is not even about responsible of 10% of this difference.

Unfortunately I do not have a Nova license to build and test a same app, with same resources, locally, on both solutions.

**Comment by tembra** [comment]
Sorry in advance @danharrin for mark you here to get your attention. But maybe you had heard something about this subject.

**Reply by danharrin** [reply]
I do not use serverless. But many people do, and the compute costs are so small that they are largely unnoticeable. I can't comment further, but I do not think you should be worrying about this until you actually hit scaling issues.
